dataprofiler/tests/profilers/test_profile_builder.py

```diff
@@ -303,6 +303,23 @@ def test_sample_size_warning_in_the_profiler(self, *mocks):
                                    "this subsample and not the whole dataset."):
             profile1 = dp.Profiler(data, samples_per_update=3)
 
+    def test_min_col_samples_used(self):
+        # No cols sampled since no cols to sample
+        empty_df = pd.DataFrame([])
+        empty_profile = dp.Profiler(empty_df)
+        self.assertEqual(0, empty_profile._min_col_samples_used)
+
+        # Every column fully sampled
+        full_df = pd.DataFrame([[1, 2, 3], [4, 5, 6], [7, 8, 9]])
+        full_profile = dp.Profiler(full_df)
+        self.assertEqual(3, full_profile._min_col_samples_used)
+
+        # First col sampled only once, so that is min
+        sparse_df = pd.DataFrame([[1, None, None], [1, 1, None], [1, None, 1]])
+        sparse_profile = dp.Profiler(sparse_df, min_true_samples=1,
```

Comment:
i think we should set min_true_samples to 2 as it is possible in this case for it to randomly select non-null sample_ids.
Additionally, this test is not isolated to just testing `test_min_col_samples_used` and requires running the entire profiler which may not be necessary.